### ISSUE TITLE
Accessing slider elements

### ISSUE THREAD

**Issue by arangas** [jssor/slider#70]
As mentioned in #69 I need to make the slider WCAG 2.0 Level A compliant.

Unfortunately the slider allows tabbing with a keyboard through slides that are not visible on the screen, which breaks the [Meaningful Sequence](http://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-sequence.html) requirement. Therefore I would like to hook into the `$EVT_PARK` event and perform the following operations:

1. On all `a` elements of each slide set `tabindex=-1` so that they *cannot* be focussed with the keyboard
2. On all `a` elements in the current slide set `tabindex=0` so they *can* be focussed with the keyboard

The event passes in `slideIndex` but I can't find a way to access the slide DOM elements themselves. Either the current slide passed into this event, or a way of accessing something like `_SlideElmts` would work. Note I am using the non-jQuery version.

Is there a way to do this?

**Comment by jssor** [comment]
```html

        <div u="slides" ...>
            <div id="slide_0"><img u="image" src="../img/landscape/01.jpg" /></div>
            <div id="slide_1"><img u="image" src="../img/landscape/02.jpg" /></div>
            <div id="slide_2"><img u="image" src="../img/landscape/03.jpg" /></div>
            <div id="slide_3"><img u="image" src="../img/landscape/04.jpg" /></div>
            <div id="slide_4"><img u="image" src="../img/landscape/05.jpg" /></div>
            <div id="slide_5"><img u="image" src="../img/landscape/06.jpg" /></div>
            <div id="slide_6"><img u="image" src="../img/landscape/07.jpg" /></div>
            <div id="slide_7"><img u="image" src="../img/landscape/08.jpg" /></div>
        </div>
```

```javascript
            function SliderParkEventHandler(slideIndex, fromIndex) {
                var slideElement = document.getElementById("slide_" + slideIndex);
                var anchorElements = document.getElementsByTagName("A");

                for (var i = 0; i < anchorElements.length; i++) {
                    anchorElements[i].setAttribute("tabIndex", 1);
                }
            }

            jssor_slider1.$On($JssorSlider$.$EVT_PARK, SliderParkEventHandler);
```

**Comment by arangas** [comment]
Thanks for that great code snippet. I wasn't setting IDs on the slide elements so my brain wasn't working in that way but now I can see your point. Unfortunately i've applied your approach and still have problems.

In the below code snippet I set all anchors to have `tabindex=-1`, then update all anchors in the current slide element to have `tabindex=0`. The former works correctly but the latter does nothing, all anchors stay with `tabindex=-1`. I've tried in IE and Firefox with two different approaches (you can see one of them commented). Hopefully I've made some dumb mistake that will be obvious to you. I have some concerns that it will work as there seems to be a copy of the slider elements in the DOM so therefore the IDs are not unique - maybe it is picking up the wrong one? Anyway, I value your advice:

```javascript
aquanetSlider.$On($JssorSlider$.$EVT_PARK, function (slideIndex) {
	var anchorElements,
		slideElement,
		anchorIndex,
		anchorElement;

	console.log('PARK: slideIndex=' + slideIndex);

	anchorElements = aquanetSlider.$Elmt.getElementsByTagName("a");
	console.log('Number of anchorElements: ' + anchorElements.length);
	for (anchorIndex = 0; anchorIndex < anchorElements.length; anchorIndex++) {
		console.log('setting tabindex of anchor with index ' + anchorIndex + ' to -1');
		anchorElement = anchorElements[anchorIndex];
//    		anchorElement.removeAttribute("tabIndex");
//    		anchorElement.setAttribute("tabIndex", "-1");
anchorElement.tabIndex = -1;
	}
	
	slideElement = document.getElementById("aquanet-slider-slide-" + slideIndex);
	anchorElements = slideElement.getElementsByTagName("a");
	console.log('Number of anchorElements: ' + anchorElements.length);
	for (anchorIndex = 0; anchorIndex < anchorElements.length; anchorIndex++) {
		console.log('setting tabindex of anchor with index ' + anchorIndex + ' in slideIndex ' + slideIndex + ' to 0');
		anchorElement = anchorElements[anchorIndex];
//    		anchorElement.removeAttribute("tabIndex");
//    		anchorElement.setAttribute("tabIndex", "0");
anchorElement.tabIndex = 0;
	}
});
```javascript

**Comment by jssor** [comment]
I tested on ie, firefox and chrome. It works OK.
